src/lib/mina_base/parties.ml

```diff
@@ -1429,6 +1429,49 @@ let dummy =
   ; memo = Signed_command_memo.empty
   }
 
+(* Parties transactions are filtered using this predicate
+   - in outgoing gossip
+   - when adding to the transaction pool
+   - in incoming blocks
+
+   Design note: a single predicate means we traverse the other parties only
+   once; the alternative would be to use separate predicates, so we could log
+   what bound is exceeded, at the expense of multiple traversals
+*)
+let valid_size t =
+  let max_proof_parties = 4 in
+  let max_nonproof_parties = 8 in
```

Comment:
I think the following rules would potentially capture real system limitations better:
* `num_proof_parties` has to be <= 4
* `num_proof_parties + num_nonproof_parties` have to be <= 8
this keeps max total parties at 8 while still giving flexibility similar to the current rule

Reply:
Done.